pkg/testutils/testcluster/testcluster.go

```diff
@@ -108,6 +108,11 @@ func (tc *TestCluster) stopServers(ctx context.Context) {
 	for i := range tc.mu.serverStoppers {
 		if tc.mu.serverStoppers[i] != nil {
 			tc.mu.serverStoppers[i].Stop(context.TODO())
+			// We need to check IsStopped() in addition to just waiting for
+			// Stop() above to complete, because if there was an asynchronous
+			// Drain request ongoing, Stop() would take a shortcut and not
+			// actually wait for the async stop to complete.
+			<-tc.mu.serverStoppers[i].IsStopped()
```

Comment:
`Stop()` should wait until `IsStopped()` is closed, though. I suggest adding a `<-s.IsStopped()` before the early return in `(*Stopper).Stop`.

Reply:
Ok done. Let's see in CI whether there are tests that rely on this 2nd-stop-is-noop behavior.

Reply:
It is a noop :-) Just a bit slower now. I doubt anything will shake out but it's good to check of course.